cpp/include/cugraph/graph_mask.hpp

```diff
@@ -0,0 +1,382 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/handle.hpp>
+#include <raft/core/span.hpp>
+#include <rmm/device_uvector.hpp>
+
+#include <cstdint>
+#include <limits>
+#include <optional>
+#include <vector>
+
+namespace cugraph {
+
+/**
+ * Compile-time fast lookup of log2(num_bits(mask_t)) to eliminate
+ * the log2 computation for powers of 2.
+ * @tparam mask_t
+ */
+template <typename mask_t>
+__host__ __device__ constexpr int log_bits()
+{
+  switch (std::numeric_limits<mask_t>::digits) {
+    case 8: return 2;
```

Comment:
Shouldn't this be 3?

Reply:
Yep that should be 3. I must have made a typo when I was cleaning this up.